app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -56,23 +47,30 @@ public void writeToParcel(Parcel dest, int flags) {
     @Override
     public Where<Message, Object> modWhere(Where<Message, Object> where)
             throws SQLException {
-        where.like(Message.TIMESTAMP_FIELD, new SelectArg(date));
+        DateUtils.truncate(new Date(), java.util.Calendar.DAY_OF_MONTH);
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        calendar.set(Calendar.HOUR_OF_DAY,0);
+        calendar.set(Calendar.MINUTE,0);
+        calendar.set(Calendar.SECOND,0);
+        calendar.set(Calendar.MILLISECOND, 0);
+        where.between(Message.TIMESTAMP_FIELD,calendar.getTime().getTime(),calendar.getTime().getTime()+86400000);
```

Comment:
Declare this 86400000 into a constant, this would help to understand this easier!